pennylane/ops/op_math/exp.py

```diff
@@ -0,0 +1,240 @@
+    >>> print(qml.draw(circuit)(1.23))
+    0: ──Exp─┤  <Z>
+
+    If the base operator is Hermitian and the coefficient is real, then the ``Exp`` operator
+    can be measured as an observable:
+
+    >>> obs = qml.exp(qml.PauliZ(0), 3)
+    >>> @qml.qnode(qml.device('default.qubit', wires=1))
+    ... def circuit():
+    ...     return qml.expval(obs)
+    >>> circuit()
+    tensor(20.08553692, requires_grad=True)
+
+    """
+    return Exp(op, coeff, id=id)
+
+
+class Exp(SymbolicOp):
+    """A symbolic operator representating the exponential of a operator.
+
+    Args:
+        base (~.operation.Operator): The Operator to be exponentiated
+        coeff=1 (Number): A scalar coefficient of the operator.
+
+    **Example**
+
+    This symbolic operator can be used to make general rotation operators:
+
+    >>> x = np.array(1.23)
+    >>> op = Exp( qml.PauliX(0), -0.5j * x)
+    >>> qml.math.allclose(op.matrix(), qml.RX(x, wires=0).matrix())
+    True
+
+    This can even be used for more complicated generators:
+
+    >>> t = qml.PauliX(0) @ qml.PauliX(1) + qml.PauliY(0) @ qml.PauliY(1)
+    >>> isingxy = Exp(t, 0.25j * x)
+    >>> qml.math.allclose(isingxy.matrix(), qml.IsingXY(x, wires=(0,1)).matrix())
+    True
+
+    If the coefficient is purely imaginary and the base operator is Hermitian, then
+    the gate can be used in a circuit, though it may not be supported by the device and
+    may not be differentiable.
+
+    >>> @qml.qnode(qml.device('default.qubit', wires=1))
+    ... def circuit(x):
+    ...     Exp(qml.PauliX(0), -0.5j * x)
+    ...     return qml.expval(qml.PauliZ(0))
+    >>> print(qml.draw(circuit)(1.23))
+    0: ──Exp─┤  <Z>
+
+    If the base operator is Hermitian and the coefficient is real, then the ``Exp`` operator
+    can be measured as an observable:
+
+    >>> obs = Exp(qml.PauliZ(0), 3)
+    >>> @qml.qnode(qml.device('default.qubit', wires=1))
+    ... def circuit():
+    ...     return qml.expval(obs)
+    >>> circuit()
+    tensor(20.08553692, requires_grad=True)
+
+    """
+
+    coeff = 1
+    """The numerical coefficient of the operator in the exponent."""
+
+    control_wires = Wires([])
+
+    def __init__(self, base=None, coeff=1, do_queue=True, id=None):
+        self.coeff = coeff
+        super().__init__(base, do_queue=do_queue, id=id)
+        self._name = "Exp"
+
+    def __repr__(self):
+        return (
+            f"Exp({self.coeff} {self.base})"
+            if isinstance(self.base, Tensor)
+            else f"Exp({self.coeff} {self.base.name})"
+        )
+
+    @property
+    def data(self):
+        return [[self.coeff], self.base.data]
+
+    @data.setter
+    def data(self, new_data):
+        self.coeff = new_data[0][0]
+        self.base.data = new_data[1]
+
+    @property
+    def num_params(self):
+        return self.base.num_params + 1
+
+    @property
+    def is_hermitian(self):
+        return self.base.is_hermitian and math.imag(self.coeff) == 0
```

Comment:
Is it okay for `is_hermitian` to report false negatives? E.g. for an anti-Hermitian operator and a purely imaginary coefficient, the exponential is Hermitian, but this function says it's not.

Reply:
@dwierichs, when we were adding this property we decided to go with "non-exhaustive" quick checks to determine hermiticity. The rational was we wanted some coverage but also efficient checks. In some cases the function will return false negatives 😢

But since we no longer rely on this property to determine if the op can be measured, maybe we can remove this @albi3ro? Then rely on the top level function which should do an exhaustive check.

Reply:
Ah thanks for the explanation @Jaybsoni 
Maybe there is a nice structure we can have that "collects positives". I.e. there is an (expensive) base version of determining Hermiticity, and then subclasses can define additional sufficient conditions, or replace the base version altogeher. The easiest structure that comes to my mind would be along the lines of
```python
@property
def is_hermitian(self):
    cheap_check = self.base.is_hermitian and math.imag(self.coeff) == 0
    return cheap_check or super().is_hermitian
```
It should be noted that we can make things worse with this, for example whenever the `self.base.is_hermitian` check calls the expensive matrix-level method, the `cheap_check` is False, and then there is a second check called for the `Exp` operator itself...